### PR TITLE
Fix dangling reference into container

### PR DESCRIPTION
**Pull request by Atlante45**
Fixes a crash in asset client

### CODE REVIEW

**Comment by Atlante45** [comment]
https://app.asana.com/0/26551373415487/117194525159205

**Comment by Atlante45** [comment]
Replicating this crash is really hard since it's a race and as an added difficulty, the map needs to decide to do a rehash.

So we'll stick to standard smoke test.
Make sure ATP resources load and reload correctly.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9a7bebaaa2a12ba5dff7305c1e3ee88c973ad729 available at: https://gist.github.com/c3c79e022dd31b8030369f4355508048<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7792.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7792.dmg

**Comment by Atlante45** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9a7bebaaa2a12ba5dff7305c1e3ee88c973ad729 available at: https://gist.github.com/c3c79e022dd31b8030369f4355508048<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7792.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7792.dmg

**Comment by kevinmthomas-carpool** [comment]
testing

**Comment by kevinmthomas-carpool** [comment]
## QA Test
**PR Build:** 7792
**Testing Architecture:** 64-bit PC Desktop
**Testing OS:** Windows 10
**Testing Locations:** localhost, playa, cellscience
**Smoke Test Results:** Successful test.

                                  |localhost |playa    |cellscience   
----------------------------------|----------|---------|-----------
Render Rate:                      |58.97     |59.99    |59.96       
Present Rate:                     |60.01     |60.00    |59.99       
Present New Rate:                 |59.04     |59.98    |59.89      
Render Stats Num Buffers:         |4603      |4193     |2708        
Render Stats gpu::Buffer Memory:  |35.5Mb    |60.2Mb   |22.8Mb      
Render Stats Num Textures:        |282       |258      |92          
Render Stats gpu::Texture Memory: |524.4Mb   |1598.5Mb |603.2Mb     
Render Stats Triangles:           |541k      |739k     |489k       
Render Stats Drawcalls:           |494       |345      |158         
Render Stats State Changes:	  |185       |257      |56          
Render Stats Items:               |333       |406      |751        
Render Stats Timing:              |31.6ms    |22.1ms   |12.9ms      
GPU-Z Memory Size                 |4096Mb    |4096Mb   |4096Mb      
GPU-Z Initial Memory		  |243Mb     |243Mb    |243Mb       
GPU-Z Memory Used		  |1706Mb    |2588Mb   |1388Mb      
GPU-Z Memory Load		  |8%        |26%      |17%        

**localhost ‘createTestBlocks.js’ Result:** Successful test.
**localhost ATP Upload (kelectricguitar10.fbx):** Successful test.
**localhost Echo Server Audio Result:** Successful test.
**localhost ‘harmonicOscillator.js’ Result:** Successful test.
**localhost ‘reverbTest.js’ Result:** Successful test.
**localhost ‘playaPerformanceTest.js’ Result:** Successful test.  Test results significantly beneath the red line.  Screen shot enclosed in attachments.
**playa ‘playaPerformanceTest.js’ Result:** Successful test.  Test results significantly beneath the red line.  Screen shot enclosed in attachments.
**cellscience ‘playaPerformanceTest.js’ Result:** Successful test.  Test results significantly beneath the red line.  Screen shot enclosed in attachments.
**Additional Test Results:** (none)
**Successful Logout/Crash:** Successful logout.
localhost performance
play performance
cellscience performance

File: libraries/networking/src/AssetClient.h [hunk 1]
```diff
@@ -74,6 +74,9 @@ private slots:
     void handleAssetGetReply(QSharedPointer<ReceivedMessage> message, SharedNodePointer senderNode);
     void handleAssetUploadReply(QSharedPointer<ReceivedMessage> message, SharedNodePointer senderNode);
 
+    void handleProgressCallback(const QWeakPointer<Node>& node, MessageID messageID, DataOffset length);
+    void handleCompleteCallback(const QWeakPointer<Node>& node, MessageID messageID);
```
Comment (huffman) [inline libraries/networking/src/AssetClient.h]: do these need to be slots?

**Comment by huffman** [comment]
Looks good other than my one comment

File: libraries/networking/src/AssetClient.h (hunk 1 above)
Comment (Atlante45) [inline libraries/networking/src/AssetClient.h]: Too big to put in an already big function.

Comment (huffman) [inline libraries/networking/src/AssetClient.h]: But are they being connected to anything? Aren't they just being called directly?

Comment (Atlante45) [inline libraries/networking/src/AssetClient.h]: Gotcha

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6d9080162cb11b7beacb9d8844f3e0ec3add4370 available at: https://gist.github.com/c3c79e022dd31b8030369f4355508048<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7792.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7792.dmg

**Pull request merged**